src/Numerics/LinearAlgebra/Storage/DenseVectorStorage.cs

```diff
@@ -115,9 +115,9 @@ public static DenseVectorStorage<T> OfVector(VectorStorage<T> vector)
 
         public static DenseVectorStorage<T> OfValue(int length, T value)
         {
-            if (length < 1)
+            if (length < 0)
             {
-                throw new ArgumentOutOfRangeException(nameof(length), string.Format(Resources.ArgumentLessThanOne, length));
+                throw new ArgumentOutOfRangeException(nameof(length), Resources.ArgumentNotNegative);
```

Comment:
You have 2 messages that are relatively close: `ArgumentNotNegative` and `ArgumentMustBePositive`. The first one explicitly mention that zero is allowed while the second doesn't say anything. When reading the message about the "must be positive" case, I wouldn't have guessed this wasn't allowing zero.

Anyways, which message do you want to use? Shall I remove the other?

Reply:
Semantically for integers I understand positive to not include zero, and non-negative to include zero. I noticed that this is explicitly mentioned in the text of `ArgumentPositive` but not in `ArgumentMustBePositive`.

Yes, unfortunately these resources are a bit of a mess currently, I need to clean them up. Or even get rid of them actually, since we don't have any translations anyway...

Reply:
Chatting with some native English speaker friends, it seems that for Maths in English zero is neither included in positives nor negatives. So saying "must be positive" should exclude zero but saying "should not be negative" does include zero.